packages/evm/test/voting/crvoting-commit.js

```diff
@@ -163,4 +171,439 @@ contract('CRVoting', ([_, voter]) => {
+    const voteIds = [voteId, voteId]
+    const principals = [voter, anotherVoter]
+
+    context('when the given vote ID is valid', () => {
+      beforeEach('create voting and mock weight', async () => {
+        await disputeManager.create(voteId, POSSIBLE_OUTCOMES)
+        await disputeManager.mockVoterWeight(voter, 10)
+        await disputeManager.mockVoterWeight(anotherVoter, 10)
+      })
+
+      beforeEach('allow representative', async () => {
+        await voting.setRepresentatives([representative], [true], { from: voter })
+        assert.isTrue(await voting.isRepresentativeOf(voter, representative), 'representative is not allowed')
+
+        await voting.setRepresentatives([representative], [true], { from: anotherVoter })
+        assert.isTrue(await voting.isRepresentativeOf(anotherVoter, representative), 'representative is not allowed')
+      })
+
+      context('when the given input is valid', () => {
+        const commitments = [hashVote(OUTCOMES.HIGH), hashVote(OUTCOMES.LOW)]
+
+        it('commits all the sent votes', async () => {
+          const receipt = await voting.commitForMany(voteIds, principals, commitments, { from })
+
+          const voterOutcome = await voting.getVoterOutcome(voteId, voter)
+          assertBn(voterOutcome, OUTCOMES.MISSING, 'voter outcome should be missing')
+
+          const anotherVoterOutcome = await voting.getVoterOutcome(voteId, anotherVoter)
+          assertBn(anotherVoterOutcome, OUTCOMES.MISSING, 'another voter outcome should be missing')
+
+          assertAmountOfEvents(receipt, VOTING_EVENTS.VOTE_COMMITTED, { expectedAmount: 2 })
+          assertEvent(receipt, VOTING_EVENTS.VOTE_COMMITTED, { index: 0, expectedArgs: { voteId, voter, commitment: hashVote(OUTCOMES.HIGH), sender: representative } })
+          assertEvent(receipt, VOTING_EVENTS.VOTE_COMMITTED, { index: 1, expectedArgs: { voteId, voter: anotherVoter, commitment: hashVote(OUTCOMES.LOW), sender: representative } })
+        })
+      })
+
+      context('when the given input is not valid', () => {
+        it('reverts', async () => {
+          await assertRevert(voting.commitForMany([voteId], principals, ['0x', '0x'], { from }), VOTING_ERRORS.INVALID_INPUTS_LENGTH)
+          await assertRevert(voting.commitForMany(voteIds, [voter], ['0x', '0x'], { from }), VOTING_ERRORS.INVALID_INPUTS_LENGTH)
+          await assertRevert(voting.commitForMany(voteIds, principals, ['0x'], { from }), VOTING_ERRORS.INVALID_INPUTS_LENGTH)
+        })
+      })
+    })
+
+    context('when the given vote ID is not valid', () => {
+      it('reverts', async () => {
+        await assertRevert(voting.commitForMany(voteIds, principals, ['0x', '0x'], { from }), VOTING_ERRORS.VOTE_DOES_NOT_EXIST)
+      })
+    })
+  })
+
+  describe('commitForWithSig', () => {
+    let voter, voterPK
+    const from = representative
+
+    before('create wallet', async () => {
+      const wallet = web3.eth.accounts.create('erc3009')
```

Comment:
Haha, maybe we can update this to `create('commitWithSig')` or something :)

Reply:
Fixed in d7e01f6be6c9eaeab8b117e4bff5e64db07a46ce